src/rebar_utils.erl

```diff
@@ -414,8 +414,14 @@ user_agent() ->
     ?FMT("Rebar/~s (OTP/~s)", [Vsn, otp_release()]).
 
 reread_config(ConfigList) ->
+    SetEnv = case version_tuple(?MODULE:otp_release()) of
+        {X, _, _} when X =< 17 ->
+            fun application:set_env/3;
+        _ ->
+            fun (App, Key, Val) -> application:set_env(App, Key, Val, [{persistent, true}]) end
+    end,
```

Comment:
I disagree with the usage of persistent changes here, especially since in the issue thread there was a discussion that changing either the path or the config would work by default, but in this case it appears both solutions were implemented at once?

I would have thought we would not have needed such a check.

Also it is much better for such cases to test for capability (i.e. `erlang:function_exprted(M,F,Arity)` than to try to trust version numbers of OTP releases and their APIs.